src/engine/FullRelationScan.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Florian Kramer (florian.kramer@mail.uni-freiburg.de)
+
+#include "FullRelationScan.h"
+
+FullRelationScan::FullRelationScan(QueryExecutionContext* qec, ScanType type,
+                                   const std::string& entity_var_name,
+                                   const std::string& count_var_name)
+    : Operation(qec),
+      _type(type),
+      _entity_var_name(entity_var_name),
+      _count_var_name(count_var_name) {}
+
+string FullRelationScan::asString(size_t indent) const {
+  std::ostringstream os;
+  for (size_t i = 0; i < indent; ++i) {
+    os << " ";
+  }
+  switch (_type) {
+    case ScanType::SUBJECT:
+      os << "FULL_RELATION_SCAN for subjects.";
+      break;
+    case ScanType::PREDICATE:
+      os << "FULL_RELATION_SCAN for predicates.";
+      break;
+    case ScanType::OBJECT:
+      os << "FULL_RELATION_SCAN for objects.";
+      break;
+  }
+  return os.str();
+}
+
+size_t FullRelationScan::getResultWidth() const { return 2; }
```

Comment:
Is this correct when there is not `COUNT`?

Reply:
The count is always returned by the operation. When it is not selected it is discarded during the json assembly.

Reply:
What about when using this as a subquery e.g. to join it with `wikibase:claim`?